__tests__/data-residency.test.js

```diff
@@ -0,0 +1,134 @@
+import { resolveDataPlaneUrl } from '../utils/utils';
+import { DEFAULT_DATAPLANE_URL } from '../utils/constants';
+
+const sourceConfigWithBothDataPlane = {
+  source: {
+    dataplanes: {
+      US: [
+        {
+          url: 'https://sample.rudderlabs.com/us',
+          default: true,
+        },
+      ],
+      EU: [
+        {
+          url: 'https://sample.rudderlabs.com/eu',
+          default: true,
+        },
+      ],
+    },
+  },
+};
+
+const sourceConfigWithUSDataPlane = {
+  source: {
+    dataplanes: {
+      US: [
+        {
+          url: 'https://sample.rudderlabs.com/us',
+          default: true,
+        },
+      ],
+    },
+  },
+};
+
+const sourceConfigWithEUDataPlane = {
+  source: {
+    dataplanes: {
+      EU: [
+        {
+          url: 'https://sample.rudderlabs.com/eu',
+          default: true,
+        },
+      ],
+    },
+  },
+};
+
+const serverUrl = 'https://sample.rudderlabs.com';
+
+test('When no sc url and no load call url default will be choosen', () => {
+  const url = resolveDataPlaneUrl({ source: {} }, undefined, undefined);
```

Comment:
You don't need to send `undefined` for arguments on the right-hand side to skip them.

Reply:
Addressed